source/grid/grid_in.cc

```diff
@@ -757,6 +757,31 @@ void GridIn<dim, spacedim>::read_ucd (std::istream &in)
   tria->create_triangulation_compatibility (vertices, cells, subcelldata);
 }
 
+namespace
+{
+  // Forward declaration of helper class
+  template <int dim>
+  class Abaqus_to_UCD;
+}
+
+template <int dim, int spacedim>
+void GridIn<dim, spacedim>::read_abaqus (std::istream &in)
+{
+  Assert (tria != 0, ExcNoTriangulationSelected());
+  Assert (dim==2 || dim==3, ExcNotImplemented());
+  AssertThrow (in, ExcIO());
+
+  // Read in the Abaqus file into an intermediate object
+  // that is to be passed along to the UCD reader
+  Abaqus_to_UCD<dim> abaqus_to_ucd;
+  const bool success_abaqus_in = abaqus_to_ucd.read_in_abaqus(in);
+  AssertThrow(success_abaqus_in == true, ExcInternalError());
```

Comment:
I think this is not actually an *internal error* (and similarly for the exception a couple of lines below). It's simply a failure to read whatever you file the user has given.

My preference would be to let the function not return anything at all, and in the place in `read_in_abaqus()` where you would return `false`, simply produce the exception right away. At that point you have a better chance to provide a more accurate error message that says what it is that's actually going wrong.